affordance_primitives/src/screw_model/screw_axis.cpp

```diff
@@ -165,4 +165,21 @@ Eigen::Isometry3d ScrewAxis::getTF(double delta_theta) const
 
   return output;
 }
+
+std::vector<Eigen::Isometry3d> ScrewAxis::getWaypoints(double theta_step, size_t num_steps)
+{
+  std::vector<Eigen::Isometry3d> output;
+
+  // Check for invalid inputs
+  if (num_steps < 1 || fabs(theta_step) < 1e-8) {
```

Comment:
Not sure about the second statement.. does it ever happen? 
How did you define that range for the theta_step.

Reply:
Maybe I should remove that, this is just checking for `theta_stop == 0`, which would put all of the returned waypoints at the same spot